content/program-testing.md

```diff
@@ -0,0 +1,534 @@
+
+As a reminder, the Solana CLI has a command that, when run, will start a full-featured, single-node cluster on the your workstation that you can then deploy programs and submit transactions to. To start the local validator, simply run `solana-test-validator`.
+
+When running a local validator, you will have to allow it to run in its own terminal window. Once it’s no longer needed, you can stop it with ctrl-c.
+
+![local validator](../assets/local-validator.png)
+
+To interact with a running local validator, you'd have to configure your Solana CLI to target your localhost.
+
+```sh
+solana config set --url localhost
+```
+
+You can even open a third terminal to monitor logs that your program generates by running `solana logs`.
+
+At this point, building and deploying your programs is the same as before, except now they are just deployed to your local computer and not Devnet. To send transactions to programs deployed locally, you’ll have to ensure that whatever client you’re submitting the transaction from is targeting your local cluster.
+
+```tsx
+// In a Typescript client
+// Targeting your local host instead of Devnet
+const RPC_ENDPOINT_URL = "http://127.0.0.1:8899"
+const commitment = 'confirmed'
+const connection = new web3.Connection(RPC_ENDPOINT_URL, commitment)
+```
+
+When you’re done with local host, don’t forget to change the RPC configuration for your CLI and client back to whatever cluster you want to target! Forgetting that you changed these to local host can cause you a lot of pain and frustration down the road.
+
+`solana config set --url devnet`
+
+`const RPC_ENDPOINT_URL = "https://api.devnet.solana.com"`
+
+Feel free to read up on the [Solana Test Validator docs](https://docs.solana.com/developing/test-validator).
+
+### How to build RPC tests
+
+As stated before, you don’t have to write your unit tests in Rust - you can actually write them in just about any language you want by deploying the program to a cluster and submitting transactions from a client for the program to process. The most common way of conducting tests like this is by deploying to a local validator and writing a client testing script in Typescript using the [Mocha testing framework](https://mochajs.org/) paired with the [Chai assertion library](https://www.chaijs.com/) (it’s important to note that you can use just about any testing framework or any language for these tests, as long as there are Solana SDKs available to use).
```

Comment:
@ixmorrow sugg: I don't think last sentence needs to be a parenthetical.

maybe something a la
> While those are the most common right now, it’s important to note that you can use just about any testing framework or any language for these tests as long as there are Solana SDKs available to use!